pep-536.txt

```diff
@@ -0,0 +1,174 @@
+PEP: 536
+Title: Final Grammar for Literal String Interpolation
+Version: $Revision$
+Last-Modified: $Date$
+Author: Philipp Angerer <phil.angerer@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 11-Dec-2016
+Python-Version: 3.7
+Post-History: 12-Dec-2016
+
+Abstract
+========
+
+PEP 498 introduced Literal String Interpolation (or “f-strings”).
+The expression portions of those literals however are subject to
+certain restrictions.  This PEP proposes a formal grammar lifting
+those restrictions, promoting “f-strings” to “f expressions” or f-literals.
+
+This PEP expands upon the f-strings introduced by PEP 498,
+so this text requires familiarity with PEP 498.
+
+Terminology
+===========
+
+This text will refer to the existing grammar as “f-strings”,
+and the proposed one as “f-literals”.
+
+Furthermore, it will refer to the ``{}``-delimited expressions in
+f-literals/f-strings as “expression portions” and the static string content
+around them as “string portions”.
+
+Motivation
+==========
+
+The current implementation of f-strings in CPython relies on the existing
+string parsing machinery and a post processing of its tokens.  This results in
+several restrictions to the possible expressions usable within f-strings:
+
+#. It is impossible to use the quote character delimiting the f-string
+   within the expression portion::
+
+    >>> f'Magic wand: { bag['wand'] }'
+                                 ^
+    SyntaxError: invalid syntax
+
+#. A previously considered way around it would lead to escape sequences
+   in executed code and is prohibited in f-strings::
+
+    >>> f'Magic wand { bag[\'wand\'] } string'
+    SyntaxError: f-string expression portion cannot include a backslash
+
+#. Comments are forbidden even in multi-line f-strings::
+
+    >>> f'''A complex trick: {
+    ... bag['bag']  # recursive bags!
+    ... }'''
+    SyntaxError: f-string expression part cannot include '#'
+
+#. Expression portions need to wrap ``':'`` and ``'!'`` in braces::
+
+    >>> f'Useless use of lambdas: { lambda x: x*2 }'
+    SyntaxError: unexpected EOF while parsing
+
+These limitations are non-obvious and counter-intuitive unless the user is
+familiar with the f-literals’ implementation details.  Furthermore,
+they serve no purpose from a language user perspective and can be lifted
+by giving f-literals a regular grammar without exceptions
+and implementing it using dedicated parse code.
+
+.. note:: put preceding paragraph into “Rationale”?
+
+Rationale
+=========
+
+.. https://mail.python.org/pipermail/python-ideas/2016-August/041727.html
```

Comment:
This link should be at the References section, and add the thread title here (Let’s make escaping in f-literals impossible)

As mentioned in PEP 12 :)
https://www.python.org/dev/peps/pep-0012/#hyperlinks